src/vmm/src/vstate/memory.rs

```diff
@@ -421,9 +516,21 @@ impl GuestMemoryExtension for GuestMemoryMmap {
     }
 
     /// Dumps all contents of GuestMemoryMmap to a writer.
-    fn dump<T: WriteVolatile>(&self, writer: &mut T) -> Result<(), MemoryError> {
+    fn dump<T: WriteVolatile + std::io::Seek>(&self, writer: &mut T) -> Result<(), MemoryError> {
         self.iter()
-            .try_for_each(|region| Ok(writer.write_all_volatile(&region.as_volatile_slice()?)?))
+            .flat_map(|region| region.slots())
+            .try_for_each(|(mem_slot, plugged)| {
+                if !plugged {
+                    writer
+                        .seek(SeekFrom::Current(
+                            i64::try_from(mem_slot.slice.len()).unwrap(),
```

Comment:
Maybe move `i64::try_from(...)` into a separate line? Otherwise this indentation is crazy.